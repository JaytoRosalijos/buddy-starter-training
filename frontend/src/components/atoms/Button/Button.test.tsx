import renderer from 'react-test-renderer';
import { Button } from '.';

describe("[Atom] Button", () => {

  it("Button renders correctly without props", () => {
    const tree = renderer.create(<Button>Click Me</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button renders correctly with props", () => {
    const tree = renderer.create(<Button 
                                    variance='primary'
                                    block
                                    disabled
                                    onClick={() => {}}
                                  >Click Me</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
