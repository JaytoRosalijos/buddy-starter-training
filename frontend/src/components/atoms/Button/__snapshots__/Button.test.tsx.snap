// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Atom] Button Button renders correctly with props 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-k83k30 ant-btn-default ant-btn-block sc-bcXHqe gzxuKf"
  disabled={true}
  onClick={[Function]}
  type="button"
  variance="primary"
>
  <span>
    Click Me
  </span>
</button>
`;

exports[`[Atom] Button Button renders correctly without props 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-k83k30 ant-btn-default sc-bcXHqe gzxuKf"
  disabled={false}
  onClick={[Function]}
  type="button"
  variance="primary"
>
  <span>
    Click Me
  </span>
</button>
`;
